### ISSUE TITLE
On the problem of combining localization network training and feature extraction networks

### ISSUE THREAD

**Issue by zzzzzxh1** [tim-rohwedder/fixed-length-fingerprint-extractors#5]
Hello, would you like to inquire about the training of this localization network? Is it trained separately? Or is it to train together with the subsequent feature extraction network? If combined with the subsequent network, how can minutia map be rotated and translated accordingly?

**Comment by tim-rohwedder** [comment]
Sorry for the delayed answer. This is probably no longer relevant to you, but the localization network is trained end-to-end (on same data and with same loss as regular network).

You would have to adjust the training loop a bit and process the image with the LocalizationNetwork before passing it as input to the regular DeepPrint architecture. But you would use exactly the same loss. It is just a single module at the beginning of the pipeline that the image goes through and the loss is automatically backpropagated to it.

In the provided weights, the module is not used (I left it out because it led to worse performance on my dataset). I think depending on the dataset (especially with poor alignment) it can have a positive impact.
